addons/mail/static/src/core/common/message_model.js

```diff
@@ -361,8 +358,8 @@ export class Message extends Record {
     }
 
     get authorName() {
-        if (this.author) {
-            return this.getPersonaName(this.author);
+        if (this.author_id || this.author_guest_id) {
+            return this.getPersonaName(this.author_id || this.author_guest_id);
```

Comment:
Note: probably for another PR, but `getPersonaName` should be renamed to `getPartnerName` and only called for `author_id` as I believe no override applies for guest.